### PR TITLE
AudioBridge plugin

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

File: lib/plugin/audio.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+PluginAbstract.prototype.onJoin = function(message) {
+  var self = this;
+  //message
+  self.proxyConnection.transactions.add(message['transaction'], function(response) {
+    self.stream = Stream.generate(message['body']['room'], self.proxyConnection);
+    logger.info('adding stream', self.stream);
+    streams.add(self.stream);
+  });
+  return Promise.resolve(message);
```
Comment (tomaszdurka) [inline lib/plugin/audio.js]: Was this to be comitted here? :panda_face:

File: lib/plugin/audio.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  return PluginAudio.super_.prototype.processMessage(message);
+};
+
+
+/**
+ * @param {Object} message
+ * @returns {Promise}
+ */
+PluginAbstract.prototype.onJoin = function(message) {
```
Comment (tomaszdurka) [inline lib/plugin/audio.js]: Oh, this is `PluginAbstract`, on purpose?

**Comment by tomaszdurka** [comment]
Hmm, what about having a small trait-like/parent plugin called PluginStreaming. And use it in Audio/Video plugins?

**Comment by vogdb** [comment]
@tomaszdurka what about having it as it is now

File: lib/plugin/audio.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      self.proxyConnection.transactions.add(message['transaction'], function(response) {
+        self.stream = Stream.generate(message['body']['id'], self.proxyConnection);
+        logger.info('adding stream', self.stream);
+        streams.add(self.stream);
+      });
+      return Promise.resolve(message);
+    }).catch(function() {
+      throw new JanusError.Unauthorized(message['transaction']);
+    });
+*/
```
Comment (tomaszdurka) [inline lib/plugin/audio.js]: Please delete commented code.

**Comment by vogdb** [comment]
@tomaszdurka I think it makes sense to merge this only after #52 is merged

**Comment by tomaszdurka** [comment]
@vogdb please merge upstream and adjust.

**Comment by vogdb** [comment]
@tomaszdurka or @fvovan please review

**Comment by vogdb** [comment]
I have a question. Should we have for Audio plugin `onCreate` handler? We have it for Video plugin stream but not Audio. However Audio plugin has `create` functionality ,i.e. janus client can create Audio streams(rooms).
cc @kris-lab @tomaszdurka

**Comment by tomaszdurka** [comment]
We should have create callback handled. Please take #77 into consideration there.

**Comment by vogdb** [comment]
@tomaszdurka this issue is quite overloaded already. Can we do `onCreate` separately cause it often painful to merge into this issue cause it changes Video plugin a lot.
So @tomaszdurka please review

**Comment by tomaszdurka** [comment]
Looks good.

I will also move some method of plugin into session and will double-review then.

**Pull request merged**